workshop/5 - Security/README.md

```diff
@@ -0,0 +1,259 @@
+1. Click on the **Access control (IAM)** menu item in the left navigation.
+1. Click the **+ Add** button. Then click **Add role assignment**.
+1. Select **App Configuration Data Reader** and click **Next**.
+1. In the **Members** tab, change **Assign access to** to **Managed identity**.
+1. Click **+ Select members**.
+1. In the **Select managed identities** blade, from the **Managed identity** drop down, select **App Service** under the **System-assigned managed identity** section.
+
+    ![Screenshot of picking the managed identity](../images/5-Security/selecting-managed-identity.png)
+
+1. Enter your web application's name in the **Select** field and then choose it. It should not appear under the **Selected members** section. Then click **Select**.
+1. The blade will disappear and you should see your web application's name under the **Members** section. Click **Review and assign**. _Note: you must wait until the user is fully added to an access policy before moving on to the next step, otherwise it will fail._
+1. Go through the same steps as above, but this time add your user account as a **App Configuration Data Reader**. _Note: you will use **User, group, or service principal** instead of **Managed identity** to search for your username to add._
+1. Click **Review and assign** to move through the rest of the tabs and apply the changes.
+
+The web application (and  you!) now has read access to Azure App Configuration. Let's update the application to use the managed identity and remove the connection string from it.
+
+1. Click the **Overview** menu item in the left navigation of the Azure App Configuration service in the portal.
+1. Copy the **Endpoint** value to the clipboard.
+1. Open the **appsettings.json** file and replace the value for `AzureUrls:AppConfiguration` with the value you copied to the clipboard.
+1. Open the `Program.cs` file and import the `Azure.Identity` library:
+    ```csharp
+    import Azure.Identity;
+    ```
+1. Replace `options.Connect(appConfigUrl);` with the following:
+
+    ```csharp
+    options.Connect(new Uri(appConfigUrl), azureCredential);
+    ```
+
+    This will use the managed identity to authenticate to Azure App Configuration. In fact, it will use whoever is logged in to authenticate, including the Azure account you associated with Visual Studio.
+
+1. Before running the application locally for testing, you will need to Authorize the endpoint identity for Powershell. Run the following at your terminal `Connect-AzAccount -UseDeviceAuthentication -AuthScope 'endpoint'`
+
+1. If you are running the application locally. You should see the message from Azure App Configuration displayed in the browser.
+
+    ![Screenshot of the sample application](../images/5-Security/sample-application.png)
+
+   > **Note** Azure Active Directory lazily updates permissions. If you receive a `403 Forbidden` error (surfaced as an exception in Visual Studio), wait for five minutes and try again.
```

Comment:
I think the term "eventual consistency" would be better than saying it lazily updates